login-Example/login-Example/ViewController.swift

```diff
@@ -117,53 +113,52 @@ class ViewController: UIViewController, InfomaniakLoginDelegate, DeleteAccountDe
-                            self.testSwapRefreshToken(apiToken: token)
-                        } else if let error = error {
-                            title = "Login error"
-                            description = error.localizedDescription
-                        }
-                        print("refreshTokenConvert \(title ?? "")\n\(description ?? "")")
-                    }
-                case .failure(let failure):
-                    DispatchQueue.main.asyncAfter(deadline: .now() + 1) {
-                        self.showAlert(title: "Error", message: failure.localizedDescription)
+        SimpleResolver.sharedResolver.removeAll()
+        // Init with non infinite refresh token
+        SimpleResolver.sharedResolver.store(factory: Factory(type: InfomaniakLoginable.self) { _, _ in
+            let clientId = "9473D73C-C20F-4971-9E10-D957C563FA68"
+            let redirectUri = "com.infomaniak.drive://oauth2redirect"
+            let login = InfomaniakLogin(config: .init(clientId: clientId, redirectURI: redirectUri, accessType: .offline))
+            return login
+        })
+        SimpleResolver.sharedResolver.store(factory: Factory(type: InfomaniakTokenable.self) { _, resolver in
+            return try resolver.resolve(type: InfomaniakLoginable.self,
+                                        forCustomTypeIdentifier: nil,
+                                        factoryParameters: nil,
+                                        resolver: resolver)
+        })
+
+        @InjectService var loginService: InfomaniakLoginable
+        @InjectService var tokenService: InfomaniakTokenable
+
+        loginService.asWebAuthenticationLoginFrom(anchor: .init(),
+                                                  useEphemeralSession: false,
+                                                  hideCreateAccountButton: true) { result in
+            switch result {
+            case .success(let success):
+                self.tokenService.getApiTokenUsing(code: success.code, codeVerifier: success.verifier) { token, error in
+                    var title: String?
+                    var description: String?
+
+                    if let token = token {
+                        title = "Login completed"
+                        description =
+                            "UserId: \(token.userId)\nToken: \(token.accessToken)\nExpires in: \(token.expiresIn ?? -1)"
+                        self.testSwapRefreshToken(apiToken: token)
+                    } else if let error = error {
+                        title = "Login error"
+                        description = error.localizedDescription
                     }
+                    print("refreshTokenConvert \(title ?? "")\n\(description ?? "")")
+                }
+            case .failure(let failure):
+                DispatchQueue.main.asyncAfter(deadline: .now() + 1) {
```

Comment:
```suggestion
                Task {
            try? await Task.sleep(nanoseconds:1_000_000_000)
```

For the sake of using Swift's structured concurrency everywhere, we could do this instead.

Reply:
I removed the delays. I don't know why they were there in the first place.